webpack.config.js

```diff
@@ -75,6 +75,10 @@ const resolve = {
 	alias: {
 		'@wordpress/api-fetch__non-shim': require.resolve( '@wordpress/api-fetch' ),
 		'@wordpress/api-fetch$': path.resolve( 'assets/js/api-fetch-shim.js' ),
+		'@wordpress/element__non-shim': require.resolve( '@wordpress/element' ),
+		'@wordpress/element$': path.resolve( 'assets/js/element-shim.js' ),
+		'@wordpress/hooks__non-shim': require.resolve( '@wordpress/hooks' ),
+		'@wordpress/hooks$': path.resolve( 'assets/js/hooks-shim.js' ),
```

Comment:
I see what you mean, but I don't think we can use externals together with this. The module uses the global to ensure that it always exports references to the same objects/functions but they're still "internal". That is, if we configured them as externals, we wouldn't be able to resolve/initialize them, we'd need to create entrypoints for them. Maybe that's we we need to do as you mentioned for 3P compat, but I don't think it would work otherwise.

Reply:
Ah right, I think I didn't fully understand the intention behind this originally, but I made sense of it now :)